contracts/beacon/Voucher.sol

```diff
@@ -263,12 +269,18 @@ contract Voucher is Initializable, IVoucher {
     }
 
     /**
-     * Retrieve the expiration timestamp of the voucher.
-     * @return expirationTimestamp The expiration timestamp.
+     * Drain balance of voucher on PoCo if it is expired.
+     * Funds are sent to the VoucherHub contract.
+     * @param amount amount to be drained
      */
-    function getExpiration() external view returns (uint256) {
+    function drain(uint256 amount) external onlyExpired {
```

Comment:
```suggestion
    function drain(uint256 amount) external onlyExpired onlyVoucherHub {
```

Reply:
Bien vu !
[Allow drain only by Hub to keep balances consistent](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/25/commits/287640b66980abe722307bb67c0c6208588e76fc)